### PR TITLE
FORMS-214 text changes

### PR DESCRIPTION
**Pull request by kakanisridhar**
# Description

Help Text changes for Details, Flash card, Html, Inset Text, List and Paragraph

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] Manual Test

# Checklist:

- [X] My changes do not introduce any new linting errors
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and versioning
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts
- [ ] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Review by jenbutongit — DISMISSED** [review]
Hi Sree - please could you write tests checking the strings are correct? Me and @bertrandmc are using jest getByText("the string to search for")

**Comment by TheSpartan1980** [comment]
Content changes are good from a QA perspective

File: designer/client/__tests__/FieldEdit.jest.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  });
+  const dataValue = { data, save: jest.fn() };
+
+  const TestComponentContextProvider = ({
+    children,
+    dataValue,
+    componentValue,
+  }) => {
+    const initComponentValue = (initialState: any) => {
+      return !!componentValue ? componentValue : initialState;
```
Comment (jenbutongit) [inline designer/client/__tests__/FieldEdit.jest.tsx:77]: you can just do this with nullish coalescing - fine as it is though
```suggestion
        return componentValue ?? initialState; //if componentValue is undefined or null (nullish), default to initialState
```

File: designer/client/__tests__/FieldEdit.jest.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    };
+    const [state, dispatch] = useReducer(
+      componentReducer,
+      initComponentState({ component: dataValue.data.pages[0].components[0] }),
+      initComponentValue
```
Comment (jenbutongit) [inline designer/client/__tests__/FieldEdit.jest.tsx:82]: If you are just testing for text changes and don't need to rely on the dispatches, you can do 
```
<ComponentContext.Provider value={{ data: mockValue, dispatch: jest.fn() }}>
```

File: designer/client/__tests__/FieldEdit.jest.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      <DataContext.Provider value={dataValue}>
+        <ComponentContext.Provider value={{ state, dispatch }}>
+          {children}
+        </ComponentContext.Provider>
+      </DataContext.Provider>
+    );
+  };
+
+  test("Help text changes", () => {
+    // - when
```
Comment (jenbutongit) [inline designer/client/__tests__/FieldEdit.jest.tsx:94]: cleanup comments pls

File: designer/client/components/ComponentListSelect/__tests__/ComponentListSelect.jest.tsx [hunk 4]
```diff
@@ -210,4 +210,32 @@ describe("ComponentListSelect", () => {
...
+        title: "abc",
+        schema: {},
+        options: {
+          required: true,
+        },
+        type: "RadiosField",
+      },
+      isNew: true,
+    };
+    const { container, debug } = render(
```
Comment (jenbutongit) [inline designer/client/components/ComponentListSelect/__tests__/ComponentListSelect.jest.tsx:228]: remove debug if you're not using it

File: designer/client/components/FieldEditors/__tests__/DetailsEdit.jest.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          items: [{ text: "An item", description: "A hint", value: "12" }],
+        },
+        {
+          name: "myOtherList",
+          title: "",
+          type: "string",
+          items: [{ text: "An item", description: "A hint", value: "12" }],
+        },
+      ],
+    }); //
```
Comment (jenbutongit) [inline designer/client/components/FieldEditors/__tests__/DetailsEdit.jest.tsx:66]: remove all the extra values you don't need, like list etc. it makes the test file huge + confusing

File: designer/client/components/FieldEditors/__tests__/DetailsEdit.jest.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      componentReducer,
+      initComponentState({ component: data.pages[0].components[0] }),
+      initComponentValue
+    );
+    return (
+      <ComponentContext.Provider value={{ state, dispatch }}>
+        {children}
+      </ComponentContext.Provider>
+    );
+  }
```
Comment (jenbutongit) [inline designer/client/components/FieldEditors/__tests__/DetailsEdit.jest.tsx:80]: remove useReducer here if you don't need it

File: designer/client/components/FieldEditors/__tests__/ParaEdit.jest.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+} from "../../../reducers/component/componentReducer";
+import { ParaEdit } from "../para-edit";
+import { Data } from "@xgovformbuilder/model";
+import { DataContext } from "../../../context";
+
+initI18n();
+
+describe("details-edit test", () => {
+  function TestComponentWithContext({ children }) {
+    let data = new Data({
```
Comment (jenbutongit) [inline designer/client/components/FieldEditors/__tests__/ParaEdit.jest.tsx:17]: remove what you don't need again

File: designer/client/components/FieldEditors/__tests__/ParaEdit.jest.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    return (
+      <DataContext.Provider value={dataValue}>
+        <ComponentContext.Provider value={{ state, dispatch }}>
+          {children}
+        </ComponentContext.Provider>
+      </DataContext.Provider>
+    );
+  }
+
+  describe("Text changes", () => {
```
Comment (jenbutongit) [inline designer/client/components/FieldEditors/__tests__/ParaEdit.jest.tsx:67]: you can unnest this. just "it should render with correct screen text"  is fine  (no need for `describe("Text changes"` )

File: designer/client/field-edit.tsx [hunk 9]
```diff
@@ -46,10 +46,10 @@ export function FieldEdit() {
           rows={2}
           label={{
             className: "govuk-label--s",
-            children: ["Help Text (optional)"],
+            children: ["Help text (optional)"],
```
Comment (jenbutongit) [inline designer/client/field-edit.tsx:49]: could you i18n this one too

File: designer/client/field-edit.tsx [hunk 10]
```diff
@@ -46,10 +46,10 @@ export function FieldEdit() {
...
           }}
           hint={{
-            children: ["Text can include HTML"],
+            children: [i18n("fieldeditors.helptext")],
```
Comment (jenbutongit) [inline designer/client/field-edit.tsx:52]: do you mind camel casing so it's easier to read these please "fieldEditor.helpText"

**Comment by kakanisridhar** [comment]
> Hi Sree - please could you write tests checking the strings are correct? Me and @bertrandmc are using jest getByText("the string to search for")

done

**Review by bertrandmc — APPROVED** [review]

**Pull request merged**